packages/mdc-form-field/_mixins.scss

```
@@ -47,23 +47,43 @@
     // stylelint-disable-next-line selector-max-type
     > label {
       @include feature-targeting-mixins.targets($feat-structure) {
-        @include rtl-mixins.reflexive-property(margin, 0, auto);
         @include rtl-mixins.reflexive-property(padding, variables.$item-spacing, 0);
 
         order: 0;
       }
     }
+
+    &:not(.mdc-form-field--space-between) {
+      > label {
+        @include feature-targeting-mixins.targets($feat-structure) {
+          @include rtl-mixins.reflexive-property(margin, 0, auto);
+        }
+      }
+    }
   }
 
   .mdc-form-field--align-end {
     // stylelint-disable-next-line selector-max-type
     > label {
       @include feature-targeting-mixins.targets($feat-structure) {
-        @include rtl-mixins.reflexive-property(margin, auto, 0);
         @include rtl-mixins.reflexive-property(padding, 0, variables.$item-spacing);
 
         order: -1;
       }
     }
+
+    &:not(.mdc-form-field--space-between) {
+      > label {
+        @include feature-targeting-mixins.targets($feat-structure) {
+          @include rtl-mixins.reflexive-property(margin, auto, 0);
+        }
+      }
+    }
+  }
+
+  .mdc-form-field--space-between {
+    @include feature-targeting-mixins.targets($feat-structure) {
+      justify-content: space-between;
+    }
   }
```

I gotten some more feedback on this PR. We don't think my original suggestion to avoid overrides is the right one since it results in an overly specific selector for the default styling. Your original approach in overriding the margin with the modifier selector looks like it makes more sense.

So final change, go ahead and revert the margin changes and add the following to your space-between selector!

```suggestion
  .mdc-form-field--space-between {
    @include feature-targeting-mixins.targets($feat-structure) {
      justify-content: space-between;
    }

    > label {
      @include feature-targeting-mixins.targets($feat-structure) {
        @include rtl-mixins.reflexive-property(margin, initial, auto);
      }
    }

    &.mdc-form-field--align-end {
      > label {
        @include feature-targeting-mixins.targets($feat-structure) {
          @include rtl-mixins.reflexive-property(margin, auto, initial);
        }
      }
    }
  }
```

---

done